### PR TITLE
week-005: Johee's solution in JavaScript.

### PR DESCRIPTION
**Pull request by joheec**
I want to say this is O(n) since max 2n. Does that make sense?

### CODE REVIEW

File: week-005-array-min-swap-2/johee.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
 function minimumSwaps(arr) {
+    // creating an object. Can edit objects with a const as long as I don't reassign the variable. Fun
     const touchedNodes = {};
```
Comment (thedeadline) [inline week-005-array-min-swap-2/johee.js:3]: TIL {} = key value obj in js. 😃

File: week-005-array-min-swap-2/johee.js [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
     let totalSwaps = 0;
     
     for (let i = 0; i < arr.length; ++i) {
+        // if the node was touched, it's already in the right spot
         if (touchedNodes[i]) {
             continue;
         }
         
+        // well, let's "touch" (check) the node to see if it's in the right spot
         touchedNodes[i] = true;
```
Comment (thedeadline) [inline week-005-array-min-swap-2/johee.js:15]: ...what..? how does =true; mean it's in the right spot?

Comment (thedeadline) [reply week-005-array-min-swap-2/johee.js:15]: what javascript wizardy is this?!

Comment (joheec) [reply week-005-array-min-swap-2/johee.js:15]: It just means that I am going to check it

**Pull request merged**